bin/publish-to-registries.sh

```diff
@@ -6,9 +6,14 @@ set -x
 
 bin/build.sh "${STACK_VERSION}"
 
-# Disable tracing temporarily to prevent logging registry tokens.
+# Disable tracing temporarily (with `set +x`) to prevent logging registry tokens.
+echo "Logging into Docker hub..."
```

Comment:
The `set -x` from above will mean both the echo command as typed, and the output from the echo command will be shown in the logs (ie: the message will look as though it's been repeated twice). To avoid that, the logging can be moved inside the subshell after the `set +x`.

Re the `echo "Done"`, the docker login command already outputs "Login Succeeded" in the successful case, so I think the "done" message can be omitted.

I think it might also be worth using a multi-line statement inside the subshell now we're adding more things in there, to make it more readable. We can even collapse both subshells into one.

So how about something like this:

```bash
(
    # Disable tracing (until the end of this subshell) to prevent logging registry tokens.
    set +x

    echo "Logging into Docker Hub..."
    echo "${DOCKER_HUB_TOKEN}" | docker login -u "${DOCKER_HUB_USERNAME}" --password-stdin

    echo "Logging into internal container registry..."
    curl -sSf --retry 3 -X POST "$ID_SERVICE_TOKEN_ENDPOINT" -d "{\"username\":\"${ID_SERVICE_USERNAME}\",\"password\":\"${ID_SERVICE_PASSWORD}\"}" \
        | jq -er ".raw_id_token" \
        | docker login "$INTERNAL_REGISTRY_HOST" -u "$INTERNAL_REGISTRY_USERNAME" --password-stdin
)
```

Reply:
To speed up this PR, I've made this change myself in f963fc5.